iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/ComponentImpl.java

```diff
@@ -4,29 +4,30 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
-package com.powsybl.iidm.network.impl;
+package com.powsybl.iidm.network.util;
 
 import com.google.common.collect.Iterables;
 import com.powsybl.iidm.network.Bus;
 import com.powsybl.iidm.network.Component;
-import com.powsybl.iidm.network.impl.util.Ref;
+import com.powsybl.iidm.network.Network;
 
+import java.lang.ref.WeakReference;
 import java.util.Objects;
 import java.util.stream.Stream;
 
 /**
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
  */
-class ComponentImpl implements Component {
+public class ComponentImpl implements Component {
```

Comment:
There is a bug in the SynchronousComponent. The filter used in getBuses() should test the synchronousComponentNumber instead of the connectedComponentNumber.
Do a small refactor to fix the bug